### PR TITLE
[UPLOAD-612] workspaces

### PR DESCRIPTION
**Pull request by krystophv**
This PR supersedes [UPLOAD-428] (pr https://github.com/tidepool-org/uploader/pull/1410) as this branch was taken from that work. To get a clearer picture of the changes from that PR to here, please see draft PR: https://github.com/tidepool-org/uploader/pull/1417 

Addresses [UPLOAD-612]

[UPLOAD-428]: https://tidepool.atlassian.net/browse/UPLOAD-428
[UPLOAD-612]: https://tidepool.atlassian.net/browse/UPLOAD-612

### CODE REVIEW

File: app/components/ClinicUserSelect.js [hunk 1]
```diff
@@ -140,9 +207,86 @@ class ClinicUserSelect extends React.Component {
...
+                {this.state.clinicDropdownOpen &&
+                <div className={styles.dropdown} onClick={(e)=>e.stopPropagation()}>
+                  <ul>
+                    {_.map(clinics,(clinic) =>
+                      <li>
+                        <div
+                          className={styles.clinicItem}
+                          onClick={()=>this.handleSwitchClinic(clinic)}>
+                            {clinic.name}
+                            {clinic.id == selectedClinicId && <CheckRoundedIcon />}
```
Comment (clintonium-119) [inline app/components/ClinicUserSelect.js:230]:
```suggestion
                            {clinic.id === selectedClinicId && <CheckRoundedIcon />}
```

File: app/components/Header.js [hunk 2]
```diff
@@ -34,6 +34,7 @@ import styles from '../../styles/components/Header.module.less';
 import logo from '../../images/Tidepool_Logo_Light x2.png';
 
 import { remote } from 'electron';
+import {selectClinic} from '../actions/sync';
```
Comment (clintonium-119) [inline app/components/Header.js:37]:
```suggestion
```
This import is not actually used.

File: test/app/actions/async.test.js [hunk 3]
```diff
@@ -3240,28 +3855,271 @@ describe('Asynchronous Actions', () => {
...
+            abc123: 'US/Mountain'
+          }
+        });
+        store.dispatch(async.goToPersonalWorkspace());
+        const actions = store.getActions();
+        expect(actions).to.deep.equal(expectedActions);
+      });
+    });
+    describe('user has no supported devices', () => {
+      test('should dispatch SELECT_CLINIC, SET_UPLOAD_TARGET_USER, SET_UPLOADS, then SET_PAGE (redirect to settings page for device selection)', () => {
```
Comment (clintonium-119) [inline test/app/actions/async.test.js:3962]:
```suggestion
      test('should dispatch SELECT_CLINIC, SET_UPLOAD_TARGET_USER, then SET_PAGE (redirect to settings page for device selection)', () => {
```

**Review by clintonium-119 — APPROVED** [review]
Approved ✔️
Nice work. A couple of very minor suggestions in the comments.
And thanks for putting up the smaller PR to simplify the review.

File: app/components/Header.js (hunk 2 above)
Comment (krystophv) [reply app/components/Header.js:37]: Ah yeah, my VSCode auto-import plugin gets overzealous sometimes. Thanks for catching it.

**Pull request merged**